linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -267,19 +267,29 @@ final class Emitter(config: Emitter.Config) {
               extractWithGlobals(classEmitter.genModuleInitializer(initializer)(
                   moduleContext, uncachedKnowledge))
             }
-        )(Position.NoPosition)
+        )
+
+        /* Flatten all the top-level js.Block's, because we temporarily use
+         * them to gather several top-level trees under a single `js.Tree`.
+         * TODO We should improve this in the future.
+         */
+        val defTrees: List[js.Tree] = defTreesIterator.flatMap {
```

Comment:
IIUC this will make the portions we cache on smaller (e.g. in ES 5.1, down to individual methods). Can you just confirm this is intended?

Reply:
Yes and no. The basic linker backend would cache on the smaller level. But the emitter's `fullClass` caches at the class level, so it doesn't change anything.

(This was the same before the flattening changes you suggested, since the flattening was done by the `js.Block` anyway.)